File: tests/sample.js
/** Sample JavaScript file for metacall-parser testing */

const fs = require('fs');
const { helper } = require('./utils');

function greet(name) {
  return `Hello, ${name}!`;
}

const add = function(a, b) {
  return a + b;
};

const multiply = (x, y) => x * y;

class Calculator {
  add(x, y) {
    return x + y;
  }

  multiply(x, y) {
    return x * y;
  }
}

class DataProcessor {
  process(data) {
    return helper(data);
  }
}

File: tests/sample.py
"""Sample Python file for metacall-parser testing"""

import os
import sys
from utils import helper

def greet(name):
    """Say hello to name."""
    return f"Hello, {name}!"

def add(a, b):
    return a + b

class Calculator:
    def add(self, x, y):
        return x + y

    def multiply(self, x, y):
        return x * y

class DataProcessor:
    def process(self, data):
        return helper(data)
